app/services/set_reject_by_default.rb

```diff
@@ -17,9 +18,21 @@ def call
     return if application_choice.reject_by_default_at.to_s == time.in_time_zone.to_s &&
               application_choice.reject_by_default_days == days
 
+    rbd_date = beyond_eoc?(time) ? eoc_rbd_date : time
+
     application_choice.update!(
-      reject_by_default_at: time,
+      reject_by_default_at: rbd_date,
       reject_by_default_days: days,
     )
   end
+
+private
+
+  def beyond_eoc?(date)
+    date >= CycleTimetable.find_closes
+  end
+
+  def eoc_rbd_date
+    0.business_days.before(CycleTimetable.find_closes).end_of_day
```

Comment:
:+1: `0.business_days.before` looks redundant but having tried it I can see it takes us to the last business day before on including the given date. So that makes sense.